dotnet/src/IntegrationTests/Connectors/OpenAI/ChatHistoryTests.cs

```diff
@@ -0,0 +1,137 @@
+    {
+        // Arrange
+        this._kernelBuilder.Services.AddSingleton<ILoggerFactory>(this._logger);
+        var builder = this._kernelBuilder;
+        this.ConfigureAzureOpenAIChatAsText(builder);
+        builder.Plugins.AddFromType<FakePlugin>();
+        var kernel = builder.Build();
+
+        OpenAIPromptExecutionSettings settings = new() { ToolCallBehavior = ToolCallBehavior.AutoInvokeKernelFunctions };
+        ChatHistory history = new();
+
+        // Act
+        history.AddUserMessage("Make me a special poem");
+        var historyBeforeJson = JsonSerializer.Serialize(history.ToList(), s_jsonOptionsCache);
+        var service = kernel.GetRequiredService<IChatCompletionService>();
+        ChatMessageContent result = await service.GetChatMessageContentAsync(history, settings, kernel);
+        history.AddUserMessage("Ok thank you");
+
+        ChatMessageContent resultOriginalWorking = await service.GetChatMessageContentAsync(history, settings, kernel);
+        var historyJson = JsonSerializer.Serialize(history, s_jsonOptionsCache);
+        var historyAfterSerialization = JsonSerializer.Deserialize<ChatHistory>(historyJson);
+        var exception = await Record.ExceptionAsync(() => service.GetChatMessageContentAsync(historyAfterSerialization!, settings, kernel));
+
+        // Assert
+        Assert.Null(exception);
+    }
+
+    private void ConfigureAzureOpenAIChatAsText(IKernelBuilder kernelBuilder)
+    {
+        var azureOpenAIConfiguration = this._configuration.GetSection("Planners:AzureOpenAI").Get<AzureOpenAIConfiguration>();
+
+        Assert.NotNull(azureOpenAIConfiguration);
+        Assert.NotNull(azureOpenAIConfiguration.ChatDeploymentName);
+        Assert.NotNull(azureOpenAIConfiguration.ApiKey);
+        Assert.NotNull(azureOpenAIConfiguration.Endpoint);
+        Assert.NotNull(azureOpenAIConfiguration.ServiceId);
+
+        kernelBuilder.AddAzureOpenAIChatCompletion(
+            deploymentName: azureOpenAIConfiguration.ChatDeploymentName,
+            modelId: azureOpenAIConfiguration.ChatModelId,
+            endpoint: azureOpenAIConfiguration.Endpoint,
+            apiKey: azureOpenAIConfiguration.ApiKey,
+            httpClient: new HttpClient(new MyHttpMessageHandler()),
+            serviceId: azureOpenAIConfiguration.ServiceId);
+    }
+
+    public class FakePlugin
+    {
+        [KernelFunction, Description("creates a special poem")]
+        public string CreateSpecialPoem()
+        {
+            return "ABCDE";
+        }
+    }
+
+    public class MyHttpMessageHandler : DelegatingHandler
+    {
+        public MyHttpMessageHandler()
+        {
+            this.InnerHandler = new HttpClientHandler();
+        }
+
+        protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+        {
+            string? requestContent = await request.Content!.ReadAsStringAsync(cancellationToken);
```

Comment:
What is the purpose of this?

Reply:
I used to understand how to request was being sent to the server, was important to spot the problem, I can remove it....